### PR TITLE
List buttons

### PR DESCRIPTION
**Pull request by Brian-Demon**
### Create ReactJS component list buttons to edit a list
- `ListButtons.js` is responsible for showing three different groups of buttons depending on what the user wants to do with a list.
    - An edit button (broken up into 7 different buttons in a button group so it's the same width as the other button groups
    - Once edit is clicked, the button group changes to three buttons: share, cancel, and delete
    - The share functionality is not operational yet (future PR) but will prompt the user to enter the email of the user they wish to 
        share that particular list with.
    - The delete button, once clicked, will change the button group into three new buttons: yes, a text prompt (disabled), and no.
        Pressing Yes will delete the list although since the list group itself is not a react component the screen currently needs to be
        refreshed in order for the change to take effect (next PR will address this).
        
**This PR resolves issue: #52**

Capture
Capture
Capture

### CODE REVIEW

File: app/javascript/components/lists/ListButtons.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    buttonGroup = 
+    <React.Fragment>
+      <div className="btn-group" role="group">
+        <button className="btn btn-warning" onClick={() => setShowEditButton(1)}></button>
+        <button className="btn btn-warning" onClick={() => setShowEditButton(1)}></button>
+        <button className="btn btn-warning" onClick={() => setShowEditButton(1)}></button>
+        <button className="btn btn-warning" onClick={() => setShowEditButton(1)}>Edit</button>
+        <button className="btn btn-warning" onClick={() => setShowEditButton(1)}></button>
+        <button className="btn btn-warning" onClick={() => setShowEditButton(1)}></button>
+        <button className="btn btn-warning" onClick={() => setShowEditButton(1)}></button>
```
Comment (cadwallion) [inline app/javascript/components/lists/ListButtons.js:20]: I'm confused about why there are 7 buttons.  The PR body says it's to do with consistent sizing, but could we instead use a CSS class that sets the appropriate width, and only have one button?

Comment (Brian-Demon) [reply app/javascript/components/lists/ListButtons.js:20]: Indeed, I will look into making that change on this PR today.

Comment (Brian-Demon) [reply app/javascript/components/lists/ListButtons.js]: Fixed

**Review by cadwallion — APPROVED** [review]
👍🏻

**Pull request merged**